### ISSUE TITLE
Authenticator names are being mangled by minification

### ISSUE THREAD

**Issue by aaroncox** [EOSIO/ual-reactjs-renderer#92]
When using the ual-reactjs-renderer within an application that uses a minification build process, the names of the authenticators are getting mangled and corrupting the local storage value for `UALLoggedInAuthType`. The value saved in local storage ends up being the minified value (like `n`) and causes session persistence to break. 

This is happening because it's currently using the constructor value which seems to be a valid target for mangling:

https://github.com/EOSIO/ual-reactjs-renderer/blob/0eaaf1ee9b7f941c92a12ee212e8bdcb8621e674/src/components/provider/UALProvider.js#L170

I discovered this bug while debugging why session persistence wasn't functioning within [eoscostarica/eos-rate](https://github.com/eoscostarica/eos-rate). This is a react-scripts application, which has built-in minification and mangling. If you log in using any authenticator on eosrate.io, you'll see the local storage value of `UALLoggedInAuthType` being set improperly and the session won't persist. 

Perhaps authenticators need to implement a string value of their name which can be used, as opposed to using the constructor's name? I don't know the appropriate solution, but figured I'd raise the issue.

**Comment by zer0blockchain** [comment]
workaround for the minification issue. 
assuming you want to know what wallet you are interacting with.

```
const lynx = new Lynx([exampleNet]);
const ledger = new Ledger([exampleNet]);
const scatter = new Scatter([exampleNet], { appName });
const anchor = new Anchor([exampleNet], { appName });
const tokenPocket = new TokenPocket([exampleNet]);

LynxUser.prototype.UalUserType = "LynxUser";
LedgerUser.prototype.UalUserType = "LedgerUser";
ScatterUser.prototype.UalUserType = "ScatterUser";
AnchorUser.prototype.UalUserType = "AnchorUser";
TokenPocketUser.prototype.UalUserType = "TokenPocketUser";
```

**Comment by jlamarr22** [comment]
~~This was fixed in #95 and released in #96. Please upgrade to v0.3.0: https://www.npmjs.com/package/ual-reactjs-renderer/v/0.3.0~~

Was partially fixed in `0.3.0` and fully fixed in `0.3.1`

**Comment by aaroncox** [comment]
Unfortunately it doesn't look like this fixes the original issue. The new `getName` interface within 0.3.0 was a step towards a fix, but the usage of the `authenticator.constructor.name` pattern in these 3 places still caused the `UALLoggedInAuthType` to get mangled in production builds:

https://github.com/EOSIO/ual-reactjs-renderer/blob/3b78fd7dafbea19b9de95dffd38522a07e8341c7/src/components/provider/UALProvider.js#L169

https://github.com/EOSIO/ual-reactjs-renderer/blob/3b78fd7dafbea19b9de95dffd38522a07e8341c7/src/components/provider/UALProvider.js#L197

https://github.com/EOSIO/ual-reactjs-renderer/blob/3b78fd7dafbea19b9de95dffd38522a07e8341c7/src/components/provider/UALProvider.js#L279

I've submitted a PR which fixes these.

**Comment by jlamarr22** [comment]
@aaroncox `0.3.1` has now been released to fix this: https://www.npmjs.com/package/ual-reactjs-renderer/v/0.3.1

**Comment by aaroncox** [comment]
Fixed in #99
